src/components/shared/formItems/EthAddress.tsx

```diff
@@ -92,26 +93,30 @@ export default function EthAddress({
           : ''
       }
     >
-      <Input
-        placeholder={'juicebox.eth / ' + constants.AddressZero}
-        type="string"
-        autoComplete="off"
-        onChange={e => onInputChange(e.target.value)}
-        value={displayValue}
-      />
-      {addressForENS?.length ? (
-        <div style={{ fontSize: '0.7rem', color: colors.text.secondary }}>
-          <CheckCircleFilled /> {addressForENS}
-        </div>
-      ) : null}
-      <Form.Item
-        name={name}
-        style={{ height: 0, maxHeight: 0, margin: 0 }}
-        rules={formItemProps?.rules ?? []} // rules weren't being applied to inner FormItem
-      >
-        {/* Hidden input allows for address value to be used in form, while visible input can display ENS name */}
-        <Input hidden type="string" autoComplete="off" />
-      </Form.Item>
+      {Boolean(!disabled) && (
```

Comment:
Good point that the name is misleading. I liked the idea of having it in one component so we can just keep passing the label and extra in, rather than alternatively (opt 2), having the label and extra separated from the `EthAddress` component. 

I've changed the prop from `disabled` to `hideInput`, what do you think?

Reply:
@johnnyd-eth WDYT about this instead: separating out the switch and the input. Should avoid adding complexity to components (in this case ETH address) where possible.

```diff
diff --git a/src/components/shared/formItems/EthAddress.tsx b/src/components/shared/formItems/EthAddress.tsx
index b356cb8f..abad5714 100644
--- a/src/components/shared/formItems/EthAddress.tsx
+++ b/src/components/shared/formItems/EthAddress.tsx
@@ -12,11 +12,9 @@ import { FormItemExt } from './formItemExt'
 export default function EthAddress({
   name,
   formItemProps,
-  hideInput,
   onAddressChange,
   defaultValue,
 }: FormItemExt & {
-  hideInput?: boolean
   onAddressChange: (address: string) => void
   defaultValue: string | undefined
 }) {
@@ -94,30 +92,26 @@ export default function EthAddress({
           : ''
       }
     >
-      {!hideInput && (
-        <>
-          <Input
-            placeholder={'juicebox.eth / ' + constants.AddressZero}
-            type="string"
-            autoComplete="off"
-            onChange={e => onInputChange(e.target.value)}
-            value={displayValue}
-          />
-          {addressForENS?.length ? (
-            <div style={{ fontSize: '0.7rem', color: colors.text.secondary }}>
-              <CheckCircleFilled /> {addressForENS}
-            </div>
-          ) : null}
-          <Form.Item
-            name={name}
-            style={{ height: 0, maxHeight: 0, margin: 0 }}
-            rules={formItemProps?.rules ?? []} // rules weren't being applied to inner FormItem
-          >
-            {/* Hidden input allows for address value to be used in form, while visible input can display ENS name */}
-            <Input hidden type="string" autoComplete="off" />
-          </Form.Item>
-        </>
-      )}
+      <Input
+        placeholder={'juicebox.eth / ' + constants.AddressZero}
+        type="string"
+        autoComplete="off"
+        onChange={e => onInputChange(e.target.value)}
+        value={displayValue}
+      />
+      {addressForENS?.length ? (
+        <div style={{ fontSize: '0.7rem', color: colors.text.secondary }}>
+          <CheckCircleFilled /> {addressForENS}
+        </div>
+      ) : null}
+      <Form.Item
+        name={name}
+        style={{ height: 0, maxHeight: 0, margin: 0 }}
+        rules={formItemProps?.rules ?? []} // rules weren't being applied to inner FormItem
+      >
+        {/* Hidden input allows for address value to be used in form, while visible input can display ENS name */}
+        <Input hidden type="string" autoComplete="off" />
+      </Form.Item>
     </Form.Item>
   )
 }
diff --git a/src/components/v2/V2Project/V2ConfirmPayOwnerModal/index.tsx b/src/components/v2/V2Project/V2ConfirmPayOwnerModal/index.tsx
index 9d452292..e65428d4 100644
--- a/src/components/v2/V2Project/V2ConfirmPayOwnerModal/index.tsx
+++ b/src/components/v2/V2Project/V2ConfirmPayOwnerModal/index.tsx
@@ -202,32 +202,37 @@ export default function V2ConfirmPayOwnerModal({
               }}
             />
           </Form.Item>
-          <FormItems.EthAddress
-            defaultValue={''}
-            name={'beneficiary'}
-            onAddressChange={beneficiary => {
-              setBeneficiary(beneficiary)
-            }}
-            hideInput={!customBeneficiaryEnabled}
-            formItemProps={{
-              extra: t`Mint the new tokens to another address?`,
-              label: (
-                <div style={{ display: 'flex' }}>
-                  <Trans>Custom token beneficiary</Trans>
-                  <Switch
-                    checked={customBeneficiaryEnabled}
-                    onChange={setCustomBeneficiaryEnabled}
-                    style={{ marginLeft: 10 }}
-                  />
-                </div>
-              ),
-              rules: [
-                {
-                  validator: validateCustomBeneficiary,
-                },
-              ],
-            }}
+          <Form.Item
+            label={
+              <>
+                <Trans>Custom token beneficiary</Trans>
+                <Switch
+                  checked={customBeneficiaryEnabled}
+                  onChange={setCustomBeneficiaryEnabled}
+                  style={{ marginLeft: 10 }}
+                />
+              </>
+            }
+            extra={<Trans>Mint the new tokens to another address.</Trans>}
           />
+
+          {customBeneficiaryEnabled && (
+            <FormItems.EthAddress
+              defaultValue={''}
+              name={'beneficiary'}
+              onAddressChange={beneficiary => {
+                setBeneficiary(beneficiary)
+              }}
+              formItemProps={{
+                rules: [
+                  {
+                    validator: validateCustomBeneficiary,
+                  },
+                ],
+              }}
+            />
+          )}
+
           {hasIssuedTokens && (
             <Form.Item label={t`Receive ERC20`}>
               <Space align="start">
```